src/esprima.ts

```diff
@@ -94,11 +94,10 @@ export function parseScript(code: string, options, delegate) {
 export function tokenize(code: string, options, delegate) {
     const tokenizer = new Tokenizer(code, options);
 
-    let tokens;
-    tokens = [];
+    const tokens: any = [];
 
     try {
-        while (true) {
+        for (; ;) {
```

Comment:
Can we put a rule option that permits `while (true)`?
IMO `for(;;)` is far less readable (it does not telegraph the semantic of "forever").